website/docs/language/upgrade-guides/1-2.mdx

```diff
@@ -0,0 +1,59 @@
+---
+page_title: Upgrading to Terraform v1.2
+description: Upgrading to Terraform v1.2
+---
+
+# Upgrading to Terraform v1.2
+
+Terraform v1.2 is a minor release in the stable Terraform v1.0 series, and so
+this release should not require any unusual upgrade steps for most users.
+
+Terraform v1.2 continues to honor
+[the Terraform v1.0 Compatibility Promises](/language/v1-compatibility-promises),
+but there are some behavior changes outside of those promises that may affect a
+small number of users, described in the following sections.
+
+* [Terraform requires Linux kernel 2.6.32 or later](#terraform-requires-linux-kernel-2-6-32-or-later)
+* [Remote servers must support TLSv1.2](#remote-servers-must-support-tlsv1-2)
+* [TLS Certificates signed with SHA-1 are no longer supported](#tls-certificates-signed-with-sha-1-are-no-longer-supported)
+* [terraform-credentials-env functionality is now built into the Terraform CLI](#terraform-credentials-env-functionality-is-now-built-into-the-terraform-cli)
+
+## Terraform requires Linux kernel 2.6.32 or later
+
+The runtime which Terraform uses no longer supports Linux kernels prior to
+2.6.32. With that in mind, the official releases of Terraform v1.2 for Linux
+require distributions using kernel 2.6.32 or later. The CLI behavior on earlier
+kernel versions is undefined.
+
+## Remote servers must support TLSv1.2
+
+When making outgoing HTTPS or other TLS connections as a client, Terraform now
```

Comment:
For changes that relate to how Terraform sends outgoing requests, I think it can help to include a follow-on paragraph to remind (or teach) readers that providers are separate programs with their own rules, since in everyday use of Terraform there are far more requests being made by providers than being made by Terraform Core.

Perhaps just a short paragraph like the following, after this one?

> This change only affects requests made by Terraform Core itself, which includes provider installation, module installation, and interactions with your configured backend. Terraform provider plugins are separate programs which have their own rules about which TLS versions are supported.

I think something similar would also apply to the "TLS Certificates signed with SHA-1 are no longer supported" and so I wonder about grouping them under a single heading "Obsolete TLS features no longer supported" which describes both the required protocol version and the removal of SHA-1 support, but perhaps that would just serve to make it harder for readers to notice that the removals apply to them, so I'm not totally sure what's best to do here.

(A possible argument in favor of the grouping is that those relying on poorly-maintained middleboxes may only know that their middlebox is running old software, not specifically what the limitations of that old software are, and so maybe "we've changed the requirements for TLS" is a better hook for them to connect the cause with the effect than having the heading directly call out protocol/hashing minutiae. :man_shrugging: )

Reply:
I think this makes sense. Re-labeled and grouped the TLS sections. @laurapacilio?